shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -0,0 +1,109 @@
+  if ([[call method] isEqualToString:@"create"]) {
+    OnCreate(call, result);
+  } else if ([[call method] isEqualToString:@"dispose"]) {
+    OnDispose(call, result);
+  } else {
+    result(FlutterMethodNotImplemented);
+  }
+}
+
+void FlutterPlatformViewsController::OnCreate(FlutterMethodCall* call, FlutterResult& result) {
+  NSDictionary<NSString*, id>* args = [call arguments];
+
+  long viewId = [args[@"id"] longValue];
+  std::string viewType([args[@"viewType"] UTF8String]);
+
+  if (views_[viewId] != nil) {
+    result([FlutterError errorWithCode:@"recreating_view"
+                               message:@"trying to create an already created view"
+                               details:[NSString stringWithFormat:@"view id: '%ld'", viewId]]);
+  }
+
+  NSObject<FlutterPlatformViewFactory>* factory = factories_[viewType].get();
+  if (factory == nil) {
+    result([FlutterError errorWithCode:@"unregistered_view_type"
+                               message:@"trying to create a view with an unregistered type"
+                               details:[NSString stringWithFormat:@"unregistered view type: '%@'",
+                                                                  args[@"viewType"]]]);
+    return;
+  }
+
+  // TODO(amirh): decode and pass the creation args.
+  views_[viewId] =
+      fml::scoped_nsobject<NSObject<FlutterPlatformView>>([[factory createWithFrame:CGRectZero
+                                                                             viewId:viewId
+                                                                            andArgs:nil] retain]);
+
+  FlutterView* flutter_view = flutter_view_.get();
+  [flutter_view addSubview:[views_[viewId].get() view]];
+  result(nil);
+}
+
+void FlutterPlatformViewsController::OnDispose(FlutterMethodCall* call, FlutterResult& result) {
+  NSDictionary<NSString*, id>* args = [call arguments];
+  int64_t viewId = [args[@"id"] longLongValue];
+
+  if (views_[viewId] == nil) {
+    result([FlutterError errorWithCode:@"unknown_view"
+                               message:@"trying to dispose an unknown"
+                               details:[NSString stringWithFormat:@"view id: '%lld'", viewId]]);
+    return;
+  }
+
+  UIView* view = [views_[viewId].get() view];
+  [view removeFromSuperview];
+  [views_[viewId] dispose];
+  views_.erase(viewId);
+  result(nil);
+}
+
+void FlutterPlatformViewsController::RegisterViewFactory(
+    NSObject<FlutterPlatformViewFactory>* factory,
+    NSString* factoryId) {
+  std::string idString([factoryId UTF8String]);
+  FML_CHECK(factories_.count(idString) == 0);
+  factories_[idString] =
+      fml::scoped_nsobject<NSObject<FlutterPlatformViewFactory>>([factory retain]);
+}
+
+void FlutterPlatformViewsController::CompositeEmbeddedView(
+    int view_id,
+    flow::EmbeddedViewCompositingParams& params) {
+  CGFloat screenScale = [[UIScreen mainScreen] scale];
```

Comment:
Can you add a threading assertion here that will trip if the threads are not the same?